app/assets/javascripts/components/course_creator/course_subject_selector.jsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import CreatableInput from '../common/creatable_input.jsx';
+
+const CourseSubjectSelector = ({ updateCourse }) => {
+  const [options, setOptions] = useState();
+
+  useEffect(() => {
+    // Will contain an array of subjects in the required object format
+    const opts = [];
+
+    fetch('/wizards/researchwrite.json')
+      .then(resp => resp.json())
+      .then((data) => {
+        const jsonObj = data;
+        // Exctracting the 5th index of the researchwrite.json file that contains all the subjects
+        // Index number may change in the future if additional information is added to researchwrite.json.  
+        // Please update as needed
+        const subjectOptions = jsonObj[5].options;
+        // itarating through all the subjects and copying to the opts array
```

Comment:
This comment isn't necessary; in general, we only need comments that explain _why_ we're doing something, not _what_ we're doing, unless it's a particularly complicated, confusing, or unusual code pattern.